application/lib/catalog.dart

```diff
@@ -117,19 +119,19 @@ class _DraggableTextState extends State<DraggableText> {
           ),
         ),
         child: Padding(
-          padding: EdgeInsets.only(top: widget.atom.parent != null ? 0.0 : 8.0),
+          padding: EdgeInsets.only(top: widget.atom!.parent != null ? 0.0 : 8.0),
           child: TextButton(
-            style: widget.atom == editor.current ? ButtonStyle(backgroundColor: MaterialStateProperty.all<Color>(Colors.yellow)) : null,
+            style: widget.atom == editor.current ? ButtonStyle(backgroundColor: WidgetStateProperty.all<Color>(Colors.yellow)) : null,
             onPressed: () {
               setState(() {
-                EditorDisposition.of(context).current = widget.atom;
+                EditorDisposition.of(context)!.current = widget.atom;
               });
             },
             child: Row(
               children: <Widget>[
                 Icon(editor.cartHolds(widget.atom) ? Icons.shopping_cart : null, size: 16.0),
-                SizedBox(width: 16.0 * widget.atom.depth + 12.0),
-                makeTextForIdentifier(context, widget.atom.identifier, widget.atom.className),
+                SizedBox(width: 16.0 * widget.atom!.depth + 12.0),
```

Comment:
if atom can't be null, we should make it non-nullable

Reply:
done